modules/vaos/spec/services/v1/fhir_service_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe VAOS::V1::FHIRService do
+  subject { VAOS::V1::FHIRService.new(user) }
+
+  let(:user) { build(:user, :vaos) }
+
+  before { allow_any_instance_of(VAOS::UserService).to receive(:session).and_return('stubbed_token') }
+
+  describe '#make_request' do
+    context 'with an invalid http method' do
+      it 'raises a no method error' do
+        expect { subject.perform(:let, 'Organization/353830') }.to raise_error(
+          NoMethodError
+        )
+      end
+    end
+
+    context 'when VAMF returns a 404' do
+      it 'raises a backend exception with key VAOS_404' do
+        VCR.use_cassette('vaos/fhir/404', match_requests_on: %i[method uri]) do
+          expect { subject.perform(:get, 'MissingResource') }.to raise_error(
+            Common::Exceptions::BackendServiceException
+          ) { |e| expect(e.key).to eq('VAOS_404') }
+        end
```

Comment:
Question: I'm a little confused why we need to have a distinct VAOS_404 exception. Are we not satisfied with the generic description returned for 404 errors?